kpops/pipeline_generator/pipeline.py

```diff
@@ -67,6 +77,55 @@ def validate_unique_names(self) -> None:
                 f"step names should be unique. duplicate step names: {', '.join(duplicates)}"
             )
 
+    def generate_graph(self) -> None:
+        for component in self.components:
+            all_input_topics = self.__get_all_input_topics(component)
+            all_output_topics = self.__get_all_output_topics(component)
+
+            component_node_name = self.__get_vertex_component_name(component)
+            self.graph.add_node(component_node_name)
+
+            self.__add_ingoing_edges(all_input_topics, component_node_name)
+            self.__add_outgoing_edges(all_output_topics, component_node_name)
+
+    def __add_outgoing_edges(
+        self, all_output_topics: list[str], component_node_name: str
+    ) -> None:
```

Comment:
I would prefer
```suggestion
    def __add_output(self, source: str, output: str) -> None:
```
`component_node_name` renamed to source to indicate the relationship
and do the for loop iteration at the callsite